src/pages/home/Home.tsx

```diff
@@ -4,23 +4,37 @@ import trashIcon from "../../assets/trash-icon.webp";
 import activateIcon from "../../assets/activate-icon.svg";
 import editIcon from "../../assets/edit-icon.png";
 import { useNavigate } from "react-router";
-import { useGetUserStudentsQuery } from "../../app/services/UserService";
 import useAuthentication from "../../hooks/useAuthentication";
 import { useDeleteStudentMutation } from "../../app/services/StudentService";
 import { ConfirmDialog } from "../../components/confirm_dialog/ConfirmDialog";
 import { useState } from "react";
+import { useGetUserStudentsQuery } from "../../app/services/UserService";
+import Filter from "../../components/filter/Filter";
+import { skipToken } from "@reduxjs/toolkit/query";
 
 function Home() {
   const { userId } = useAuthentication();
-  const { data: students } = useGetUserStudentsQuery(userId!);
   const [deleteStudent] = useDeleteStudentMutation();
   const navigate = useNavigate();
+  const [filter, setFilter] = useState("");
+  const [debouncedFilter, setDebouncedFilter] = useState("");
   const [confirmDialogOpen, setConfirmDialogOpen] = useState(false);
   const [studentToDelete, setStudentToDelete] = useState<{
     id: string;
     name: string;
   } | null>(null);
 
+  const {
+    data: students,
+    error,
+    isLoading,
+  } = useGetUserStudentsQuery(
+    userId ? { userId, filter: debouncedFilter } : skipToken
```

Comment:
Y acá usarías directamente filter


```suggestion
    userId ? { userId, filter } : skipToken
```